bokehjs/src/coffee/common/canvas.coffee

```diff
@@ -43,7 +67,7 @@ class CanvasView extends ContinuumView
                           @ctx.mozBackingStorePixelRatio ||
                           @ctx.msBackingStorePixelRatio ||
                           @ctx.oBackingStorePixelRatio ||
-                          @ctx.backingStorePixelRatio || 1
+                          @ctx.backingStorePixelRatio || 1  # ak: wtf?
```

Comment:
It just makes 1 the default if none of those attributes exist. c.f. http://jsfiddle.net/6sewxex9/